packages/venia-ui/lib/components/CheckoutPage/checkoutPage.js

```diff
@@ -0,0 +1,190 @@
+
+const GuestCheckoutOptions = props => {
+    const { classes, handleSignIn } = props;
+
+    return (
+        <div className={classes.signin_container}>
+            <Button
+                className={classes.sign_in}
+                onClick={handleSignIn}
+                priority="high"
+            >
+                {'Login and Checkout Faster'}
+            </Button>
+        </div>
+    );
+};
+
+const GreetingMessage = props => {
+    const { isGuestCheckout, classes } = props;
+
+    return (
+        <div className={classes.heading_container}>
+            <h1 className={classes.heading}>
+                {isGuestCheckout ? 'Guest Checkout' : 'Review and Place Order'}
+            </h1>
+        </div>
+    );
+};
+
+const EmptyCartMessage = props => {
+    const { classes, isGuestCheckout } = props;
+
+    return (
+        <div className={classes.signin_container}>
+            <div className={classes.heading_container}>
+                <h1 className={classes.heading}>
+                    {isGuestCheckout ? 'Guest Checkout' : 'Checkout'}
+                </h1>
+            </div>
+            <h3>There are no items in your cart.</h3>
+        </div>
+    );
+};
+
+export default props => {
+    const { classes: propClasses } = props;
+    const talonProps = useCheckoutPage({
+        createCartMutation: CREATE_CART_MUTATION,
+        getCartDetailsQuery: GET_CART_DETAILS_QUERY
+    });
+    const [
+        {
+            isGuestCheckout,
+            isCartEmpty,
+            shippingInformationDone,
+            shippingMethodDone,
+            paymentInformationDone,
+            orderPlaced
+        },
+        {
+            handleSignIn,
+            setShippingInformationDone,
+            setShippingMethodDone,
+            setPaymentInformationDone,
+            placeOrder
+        }
+    ] = talonProps;
+
+    const classes = mergeClasses(defaultClasses, propClasses);
+
+    return (
+        <div className={classes.root}>
+            <Title>{`Checkout - ${STORE_NAME}`}</Title>
+            {!isCartEmpty ? (
+                <Fragment>
+                    {isGuestCheckout && (
+                        <GuestCheckoutOptions
+                            classes={classes}
+                            handleSignIn={handleSignIn}
+                        />
+                    )}
+                    <GreetingMessage
+                        isGuestCheckout={isGuestCheckout}
+                        classes={classes}
+                    />
+                    <div className={classes.body}>
+                        <div className={classes.shipping_information_container}>
+                            <ShippingInformation
+                                onSave={setShippingInformationDone}
+                                doneEditing={shippingInformationDone}
+                            />
+                        </div>
+                        {shippingInformationDone && (
```

Comment:
Just noting that I don't personally like inline conditional renders but I'm guessing you and @jimbo already talked about it so I leave it to you guys to hash out.

Reply:
I am not sure if we have a standard for this, but I found both `if-else` and `&&` for conditional renders in our code base.

Reply:
FWIW, we settled on the more explicit ternary syntax like:

`{isAlive ? (
    <Thing />
) : null}`

Over the shorter but less familiar (I don't even know what this syntax is called):

`{isAlive && (
    <Thing />
)}`

Reply:
There is no name as such in react but `&&` or `||` are short circuit evaluation elements in CS. So I believe we can call it short circuit `short circuit rendering`.

Reply:
I've covered this subject a number of times in various PRs, but I suppose the real problem is that the pattern isn't clear because we haven't been perfectly consistent in the codebase. 😅 

1. Avoid multiline statements in JSX expressions. The main reason to use JSX is because trees are very readable (like XML).
2. Avoid short-circuiting JSX elements with `&&`. The alternative to a JSX element should be `null`—not `undefined`, not `false`, and not anything else. As Brendan said, use a ternary to be explicit about it.

I've opened #2200 to correct all of the cases in the codebase where we were inconsistent about this.